### ISSUE TITLE
Why is BBA not following agreements

### ISSUE THREAD

**Issue by ThorvaldAagaard** [EdwardPiwowar/BBA#548]
00099B3C32F51A18666CAF721F80
![image](https://github.com/EdwardPiwowar/BBA/assets/10560652/3e73c3df-bc61-411c-b6b2-7370e4e936bb)

3N is better with a 4333 even without stoppers in all suits

![image](https://github.com/EdwardPiwowar/BBA/assets/10560652/9c9ebfd8-dbd2-4ec0-acf4-2faabc79f418)

**Comment by ThePokerDude** [comment]
I have an idea how to solve it.
Technically BBA followed the agreement as the desription of 2d matches the hand.
For humans it's obvious that in such situations we should choose the more specific bid.
In BBA we have to define this.
If multiple bids are possible we should check how well they fit the actual hand.
It is possible that one of the bids matches one aspect very well (for example points)
but another aspect not that well.
So let's try to define a metric:

points: we take the possible range, but assign a maximum of (let's say 4)
each suit: we take the range of the number of the cards in that suit, but assign a maximum of (let's say 3)

so let's do the calculation

3nt bid:
points: we have a range of 2 points - score = 2
clubs: 3-4 - we have a range of 1 - score = 1
diamonds, score = 1 (see clubs)
hearts, score =1 (see clubs)
spades: exactly 3 - we have a range of 0 - score = 0

Total score = 5

2d bid (version showing fit): 
points: 10+, the range is larger than 4 points, score = 4
clubs: 0-6, the range is larger than 3 clubs, score = 3
diamonds: 4+, the range is larger than 3 diamonds, score = 3
hearts: 0-4, the range is larger than 3 hearts, score = 3
spades: 3+, the range is larger than 3 spades, score = 3

Total score = 16

As we look for the bid with the lowest score, 3nt is the clear winner.

Of course this needs refinement but I think the idea is clear. 
This method is generally applicable and not bound to a specific situation.
It could also resolve the problem - when to show a 2nd suit and when to repeat the original suit.

![image](https://github.com/user-attachments/assets/5046b38a-2d56-42d6-b9e1-9087089cc34a)
![image](https://github.com/user-attachments/assets/549a78eb-c0c4-4192-9bee-4b0a655580b7)

**Comment by EdwardPiwowar** [comment]
I agree, but I don't think it can be coded like that because of the large number of additional conditions, e.g. what our potential game is. For now, BBA tries to bid in a specific order.

**Comment by ThorvaldAagaard** [comment]
I think it is an interesting idea, but it should be priority instead. Now when BBA evaluates in a specific order, that is a sort of priority, as BBA pick the first matching.

Now if BBA implements priority instead, then it would be possible to use the principles by Andreas to calculate the priority for a bid.

So we define Pass as 11- HCP, priority 0
and 
3D as 11- HCP, 6+ D and priority 4

We could have fixed priority for some bids, and for other create a function to calculate the value.
